### PR TITLE
Adds putting proxy files in a proxy-specific folder. Closes #40

### LINKED ISSUE #40: Put proxy-specific files in a proxy-specific folder in ~/.config
Right now, we put the rootCert.pfx file in the `~/.config` folder which is a generic folder that many apps on macOS use to store their config. By looking at the file, it's not clear that it belongs to the proxy, and could lead to folks deleting it. We should see if it's possible to adjust the config path to `~/.confg/msgraph-developer-proxy/` so that it's clear which files belong to the proxy and are safe to delete after uninstalling the proxy.

### PR DESCRIPTION
**Pull request by waldekmastykarz**
Adds putting proxy files in a proxy-specific folder. Closes #40

Because the config path isn't exposed outside of the ProxyServer, implemented custom certificate cache handler where we can specify the config path including an app-specific folder.

### CODE REVIEW

**Review by garrytrinder — APPROVED** [review]
Looks good to me and I can see that you've already updated the [Uninstall](https://github.com/microsoftgraph/msgraph-developer-proxy/wiki/Uninstall) page in the wiki 👍

**Comment by waldekmastykarz** [comment]
I have't updated the Wiki yet. That said, it's a good reminder that we should after we merge this PR

File: msgraph-developer-proxy/CertificateDiskCache.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                rootCertificatePath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), proxyConfigurationFolderName);
+            }
+            else if (RunTime.IsMac) {
+                rootCertificatePath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), proxyConfigurationFolderName);
+            }
+            else {
+                var assemblyLocation = GetType().Assembly.Location;
+
+                // dynamically loaded assemblies returns string.Empty location
+                if (assemblyLocation == string.Empty) assemblyLocation = Assembly.GetEntryAssembly().Location;
```
Comment (gavinbarron) [inline msgraph-developer-proxy/CertificateDiskCache.cs:98]: Do we need to use a safe accessor here?

File: msgraph-developer-proxy/CertificateDiskCache.cs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+        var certPath = Path.Combine(path, DefaultCertificateDirectoryName);
+        if (create && !Directory.Exists(certPath)) Directory.CreateDirectory(certPath);
+
+        return certPath;
+    }
+
+    private string GetRootCertificateDirectory(bool create) {
+        if (rootCertificatePath == null) {
+            var proxyConfigurationFolderName = "msgraph-developer-proxy";
```
Comment (gavinbarron) [inline msgraph-developer-proxy/CertificateDiskCache.cs:83]: If we're keeping other path fragments as const elements at the top of the file perhaps this belongs with them?

File: msgraph-developer-proxy/CertificateDiskCache.cs (hunk 1 above)
Comment (waldekmastykarz) [reply msgraph-developer-proxy/CertificateDiskCache.cs:98]: Good catch. Fixed

**Review by gavinbarron — APPROVED** [review]

**Comment by garrytrinder** [comment]
I updated the [wiki page](https://github.com/microsoftgraph/msgraph-developer-proxy/wiki/Uninstall) with the new path.

**Pull request merged**